storage/store_pool.go

```diff
@@ -218,6 +218,18 @@ func (sp *StorePool) start(stopper *stop.Stopper) {
 	})
 }
 
+// GetStoreDescriptor returns the store detail for the given storeID.
+func (sp *StorePool) getStoreDetail(storeID proto.StoreID) *storeDetail {
```

Comment:
`storeDetails` are mutable; this should probably return a copy instead of a pointer for thread safety.